src/vertex_pulling/phase.rs

```diff
@@ -0,0 +1,66 @@
+use std::{cmp::Reverse, ops::Range};
+
+use bevy::{ecs::entity::Entity, render::{render_phase::{CachedRenderPipelinePhaseItem, DrawFunctionId, PhaseItem}, render_resource::CachedRenderPipelineId}, utils::{nonmax::NonMaxU32, FloatOrd}};
+
+pub struct AabbOpaque3d {
+    pub distance: f32,
+    pub pipeline: CachedRenderPipelineId,
+    pub entity: Entity,
+    pub draw_function: DrawFunctionId,
+    pub batch_range: Range<u32>,
+    pub dynamic_offset: Option<NonMaxU32>,
+}
+
+impl PhaseItem for AabbOpaque3d {
+    // NOTE: Values increase towards the camera. Front-to-back ordering for opaque means we need a descending sort.
+    type SortKey = Reverse<FloatOrd>;
+
+    #[inline]
+    fn entity(&self) -> Entity {
+        self.entity
+    }
+
+    #[inline]
+    fn sort_key(&self) -> Self::SortKey {
+        Reverse(FloatOrd(self.distance))
+    }
+
+    #[inline]
+    fn draw_function(&self) -> DrawFunctionId {
+        self.draw_function
+    }
+
+    #[inline]
+    fn sort(items: &mut [Self]) {
+        // Key negated to match reversed SortKey ordering
+        items.sort_by_key(|item| item.sort_key());
+        // radsort::sort_by_key(items, |item| -item.distance);
```

Comment:
Iirc radsort is faster for this (it's what bevy used before), and bevy already depends on it, but it isn't re-exported. Probably fine to just add it as a crate level dependency if adding a new pipeline is the right way forward.